servers/iceberg-fixtures/src/main/java/org/projectnessie/iceberg/metadata/NessieIceberg.java

```diff
@@ -0,0 +1,227 @@
+/*
+ * Copyright (C) 2022 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.iceberg.metadata;
+
+import static org.apache.iceberg.types.Types.NestedField.required;
+import static org.apache.iceberg.view.IcebergBridge.historyEntry;
+import static org.apache.iceberg.view.IcebergBridge.parseJsonAsViewVersionMetadata;
+import static org.apache.iceberg.view.IcebergBridge.viewVersionMetadataToJson;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import org.apache.iceberg.NullOrder;
+import org.apache.iceberg.PartitionSpec;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.Snapshot;
+import org.apache.iceberg.SortDirection;
+import org.apache.iceberg.SortOrder;
+import org.apache.iceberg.TableMetadata;
+import org.apache.iceberg.TableMetadataParser;
+import org.apache.iceberg.io.FileIO;
+import org.apache.iceberg.types.Type;
+import org.apache.iceberg.types.Types;
+import org.apache.iceberg.types.Types.NestedField;
+import org.apache.iceberg.types.Types.StructType;
+import org.apache.iceberg.view.BaseVersion;
+import org.apache.iceberg.view.Version;
+import org.apache.iceberg.view.VersionSummary;
+import org.apache.iceberg.view.ViewDefinition;
+import org.apache.iceberg.view.ViewVersionMetadata;
+
+public final class NessieIceberg {
+  static final Type[] PARTITION_PRIMITIVE_TYPES = {
+    Types.LongType.get(),
+    Types.IntegerType.get(),
+    Types.StringType.get(),
+    Types.BinaryType.get(),
+    Types.DateType.get(),
+    Types.TimeType.get(),
+    Types.UUIDType.get(),
+    Types.TimestampType.withZone(),
+    Types.TimestampType.withoutZone()
+  };
+  static final Type[] PRIMITIVE_TYPES = {
+    Types.DoubleType.get(),
+    Types.LongType.get(),
+    Types.FloatType.get(),
+    Types.IntegerType.get(),
+    Types.StringType.get(),
+    Types.BinaryType.get(),
+    Types.BooleanType.get(),
+    Types.DateType.get(),
+    Types.TimeType.get(),
+    Types.UUIDType.get(),
+    Types.TimestampType.withZone(),
+    Types.TimestampType.withoutZone()
+  };
+
+  // Deterministic w/ constant seed
+  static final Random RANDOM = new Random(42L);
```

Comment:
Well, it's still reused across invocations of the various data-generating methods so each generation is still volatile based on the total order of tests :thinking: WDYT about using a fresh fixed-seed random for each top-level generation method call?